### DISCUSSION TITLE
Rust/Yew Noob trying to deploy my fork of zzhack's Blog/Site

### DISCUSSION THREAD

**Discussion by wock9000** [zzhack-stack/zzhack#30] [Q&A] [answered]
Hi Misty and everyone! 

I've managed to use the `zzhack` template for my own personal blog (and will be giving major kudos to zzhack team for this amazing repo!)

I'm currently at the stage of deployment - as in, I want to have my Yew Frontend deployed somewhere. I was wondering if there was an easy way to do this? 

- I noticed a `vercel.json` file in the repo, but am unsure how to get it working, since Rust/WASM doesn't play as nicely with Vercel as Next.js. 

- My current approach is to try an actix-server of some sort, but I'm learning as I'm going :)

Any tips and tricks to get this deployed would be sincerely appreciated! Thanks zzhacks team :)

(Of course, willing to provide any information required, in case this comment wasn't very descriptive)

Sincerely,
Anand

**Comment by wock9000** [comment]
Here's a link to my fork at the moment! Quite a bit of work to go, but I'm happy with the progress I've made so far 😄 

https://github.com/andnasnd/andnasnd

**Comment by mistricky** [comment, marked as answer]
Hey! Andnasnd thank u for following zzhack and forked this repo to create ur own site, I'm glad someone make zzhack's code work :D

The `vercel.json` is used to be parsed by [Vercel](https://vercel.com/), and the Vercel will deploy it in a serverless container I guess. But as u see, the Vercel is not provided templates like Yew/Rust or something, but Vercel still can easier to deploy static files like HTML/CSS/JS, so u need to run `trunk build` to build dist files before, and try to deploy these dists files on Vercel.

However, having trigger build action(trunk build) before each deployment is a pain in the ass, so I put these actions(just like install deps -> build -> deploy) on CI, as u can see the `.github/workflows` that describe this things. U can create ur own version of `.github/workflows/deploy-side.yml` or use ur favorite CI platform.

--
Since the zzhack was published, A few people asked me to provide a template of the zzhack that they can use this template to create their own websites, I think providing a common template is a good thing that I wanna do, but I was thinking about how to do that for other developers can easier to use.

Actually, I'm working on [zzhack-cli](https://github.com/zzhack-stack/zzhack-cli) lol, which can fast generate a static site using zzhack template with just a few commands, but it's experimental, you can use it carefully. Also if u want, u can contribute PR or ur ideas to get it ready for release! lol 

Hope the above information can help u.

Mist

**Reply by wock9000** [reply]
Thank you so much Mist! This was extremely helpful. Let me dig into this more, and if I stumble and can't unblock myself, I'll be back!

**Reply by wock9000** [reply]
Sorry Mist! Haven't had much time to dive deeper into this, I've been working on some other stuff. I'll close the discussion and will re-open it if I have any further questions! 

Sincerely appreciate the help and detailed insight. It means a lot to me!
